In [1]:
import h2o

Dependencies: Java version 8 

In [2]:
h2o.init()

Checking whether there is an H2O instance running at http://localhost:54321..... not found.
Attempting to start a local H2O server...
  Java Version: java version "1.8.0_171"; Java(TM) SE Runtime Environment (build 1.8.0_171-b11); Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, mixed mode)
  Starting server from /Users/raj/anaconda3/lib/python3.6/site-packages/h2o/backend/bin/h2o.jar
  Ice root: /var/folders/gl/wb7rqfgd3v708q5m0ydj3pkc0000gq/T/tmpe4jovjkb
  JVM stdout: /var/folders/gl/wb7rqfgd3v708q5m0ydj3pkc0000gq/T/tmpe4jovjkb/h2o_raj_started_from_python.out
  JVM stderr: /var/folders/gl/wb7rqfgd3v708q5m0ydj3pkc0000gq/T/tmpe4jovjkb/h2o_raj_started_from_python.err
  Server is running at http://127.0.0.1:54321
Connecting to H2O server at http://127.0.0.1:54321... successful.


--------------------------  ----------------------------------------
H2O cluster uptime:         03 secs
H2O cluster timezone:       Asia/Kolkata
H2O data parsing timezone:  UTC
H2O cluster version:        3.20.0.1
H2O cluster version age:    12 days
H2O cluster name:           H2O_from_python_raj_rwwksq
H2O cluster total nodes:    1
H2O cluster free memory:    3.556 Gb
H2O cluster total cores:    8
H2O cluster allowed cores:  8
H2O cluster status:         accepting new members, healthy
H2O connection url:         http://127.0.0.1:54321
H2O connection proxy:
H2O internal security:      False
H2O API Extensions:         XGBoost, Algos, AutoML, Core V3, Core V4
Python version:             3.6.5 final
--------------------------  ----------------------------------------

In [3]:
url = "http://h2o-public-test-data.s3.amazonaws.com/smalldata/iris/iris_wheader.csv"
iris = h2o.import_file(url)

Parse progress: |█████████████████████████████████████████████████████████| 100%


150 rows in iris df - say 120 - for training model and 30 for test

In [4]:
iris.nrows

150

In [5]:
train,test= iris.split_frame( [0.8] )

In [9]:
train.nrows

120

In [10]:
test.nrows

30

Go to h2o flow to inspect in getFrames to see datasets generated with respective sizes

In [11]:
from h2o.estimators.random_forest import H2ORandomForestEstimator

In [13]:
mRF=H2ORandomForestEstimator()
mRF.train( ['sepal_len','sepal_wid','petal_len','petal_wid'] , "class", train)

drf Model Build progress: |███████████████████████████████████████████████| 100%


In [14]:
mRF

Model Details
H2ORandomForestEstimator :  Distributed Random Forest
Model Key:  DRF_model_python_1529402118725_2


ModelMetricsMultinomial: drf
** Reported on train data. **

MSE: 0.04007896637134224
RMSE: 0.20019731859178894
LogLoss: 0.12613033667977022
Mean Per-Class Error: 0.05012531328320802
Confusion Matrix: Row labels: Actual class; Column labels: Predicted class



Iris-setosa    Iris-versicolor    Iris-virginica    Error      Rate
-------------  -----------------  ----------------  ---------  -------
40             0                  0                 0          0 / 40
0              35                 3                 0.0789474  3 / 38
0              3                  39                0.0714286  3 / 42
40             38                 42                0.05       6 / 120

Top-3 Hit Ratios: 


k    hit_ratio
---  -----------
1    0.95
2    1
3    1

Scoring History: 


timestamp            duration    number_of_trees    training_rmse        training_logloss     training_classification_error
---  -------------------  ----------  -----------------  -------------------  -------------------  -------------------------------
     2018-06-19 15:26:26  0.001 sec   0.0                nan                  nan                  nan
     2018-06-19 15:26:26  0.011 sec   1.0                0.3219604581083923   3.386542750248846    0.07317073170731707
     2018-06-19 15:26:26  0.015 sec   2.0                0.21896393648144905  1.4383921033678349   0.0410958904109589
     2018-06-19 15:26:26  0.019 sec   3.0                0.238059896261394    1.567878224764822    0.043478260869565216
     2018-06-19 15:26:26  0.023 sec   4.0                0.22734069882259175  1.4035479307545178   0.038834951456310676
---  ---                  ---         ---                ---                  ---                  ---
     2018-06-19 15:26:26  0.163 sec   46.0               0.20561656505695544  0.1355913122908223   0.05
     2018-06-19 15:26:26  0.166 sec   47.0               0.20412658674759349  0.13151056983969894  0.05
     2018-06-19 15:26:26  0.169 sec   48.0               0.20370506232157926  0.13139655429141986  0.05
     2018-06-19 15:26:26  0.171 sec   49.0               0.2023875407689835   0.12915253602098628  0.05
     2018-06-19 15:26:26  0.174 sec   50.0               0.20019731859178894  0.12613033667977022  0.05


See the whole table with table.as_data_frame()
Variable Importances: 


variable    relative_importance    scaled_importance    percentage
----------  ---------------------  -------------------  ------------
petal_len   1566.08                1                    0.451107
petal_wid   1535.46                0.980449             0.442287
sepal_len   284.715                0.181801             0.0820118
sepal_wid   85.3817                0.0545194            0.0245941

Scoring history shows how many trees it used which is 50 in 0.174 sec, 6 errrors in training data which is 95% accuracy 

model performance: 94 percent hit ratio data which is seen

In [19]:
p=mRF.predict(test)

drf prediction progress: |████████████████████████████████████████████████| 100%


Confidence in prediction

In [20]:
p

predict Iris-setosa Iris-versicolor Iris-virginica 
 
 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997474 0 0.00252551 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474 
 Iris-setosa 0.997575 0 0.00242474

In [21]:
mRF.model_performance(test)


ModelMetricsMultinomial: drf
** Reported on test data. **

MSE: 0.018397692132444212
RMSE: 0.13563809248306397
LogLoss: 0.06182380655459587
Mean Per-Class Error: 0.08333333333333333
Confusion Matrix: Row labels: Actual class; Column labels: Predicted class



Iris-setosa    Iris-versicolor    Iris-virginica    Error      Rate
-------------  -----------------  ----------------  ---------  ------
10             0                  0                 0          0 / 10
0              12                 0                 0          0 / 12
0              2                  6                 0.25       2 / 8
10             14                 6                 0.0666667  2 / 30

Top-3 Hit Ratios: 


k    hit_ratio
---  -----------
1    0.933333
2    1
3    1

model performance: 93 percent hit ratio data which is unseen .ie only 2 mistakes

In [23]:
#help(h2o.estimators.random_forest)

In [49]:
h2o.cluster().shutdown()

H2O session _sid_9553 closed.
